### PR TITLE
chore: add game category to user menu

### PR DESCRIPTION
**Pull request by Chef-Yogi**
### Summary
🖱️🥞🎮

This pull request adds a new menu item for the Pancake Protectors game, with a custom icon and a badge to highlight the new feature. It also refactors some code related to the menu components and styles to improve readability and user experience. It affects the files `config.ts`, `useMenuItems.ts`, `DropdownMenu.tsx`, `Menu.tsx`, `styles.tsx`, `PancakeProtector.tsx`, and `index.tsx`.

> _We are the Pancake Protectors, we fight for the sweetest prize_
> _We have a new icon and a badge, we'll make the enemies cry_
> _We click the menu items, they show us the way to glory_
> _We sort the links for mobile, we'll write our own story_

### Walkthrough
*  Add a new menu item for the Pancake Protectors game with a custom icon and a badge ([link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR7), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR165-R176), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-b86c9302201027cdb305dc610167684031e184d1c81fb76659485559f3ca09aeR32), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-b86c9302201027cdb305dc610167684031e184d1c81fb76659485559f3ca09aeR67-R70), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-d5748bbba35280005dd7f9dae278f09e2205045d50e8230a00c8f9fd24d73225L97-R98), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-d5748bbba35280005dd7f9dae278f09e2205045d50e8230a00c8f9fd24d73225R106-R111), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-d5748bbba35280005dd7f9dae278f09e2205045d50e8230a00c8f9fd24d73225R153), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-856c77d332af00634c4390334c858b845394d2a9c745e6c541e396a857833751R1-R84), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-0fc38a6b27ab3a4a4deb81972f9c6fbdfaccf9abb6231fbd89c6c12a3448f7bfR176))
*  Remove the redundant game link from the More sub-menu ([link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adL202-L206))
*  Sort the menu items for the mobile view to avoid badge overlap ([link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-f1d51a3e8a3d5ebab85da832561177073051362276bc2305a1b715566b06dc53R100-R114), [link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-f1d51a3e8a3d5ebab85da832561177073051362276bc2305a1b715566b06dc53L151-R166))
*  Add a pointer cursor style to the menu items ([link](https://github.com/pancakeswap/pancake-frontend/pull/7027/files?diff=unified&w=0#diff-dcd8a19d8a0d4edc5aa9e796bd3caf6c1257ca3d073554b48d7c535d6b8ad474R6))

### CODE REVIEW

File: apps/web/src/components/Menu/config/config.ts [hunk 1]
```diff
@@ -160,6 +161,19 @@ const config: (
...
+    {
+      label: t('Game'),
+      icon: PancakeProtectorIcon,
+      supportChainIds: SUPPORT_ONLY_BSC,
+      hideSubNav: true,
+      items: [
+        {
+          label: t('Pancake Protectors'),
+          href: 'https://protectors.pancakeswap.finance',
+          type: DropdownMenuItemType.EXTERNAL_LINK,
```
Comment (ChefMomota) [inline apps/web/src/components/Menu/config/config.ts:173]: Add
`status: { text: t('New'), color: 'success' }`

File: apps/web/src/components/Menu/hooks/useMenuItems.ts [hunk 2]
```diff
@@ -63,6 +64,10 @@ export const useMenuItems = (onUsCitizenModalPresent?: () => void): ConfigMenuIt
             const href = `${innerItem.href}${multiChainPaths[chainId] ?? ''}`
             return { ...innerItem, href }
           }
+          if (isPanCakeProtector) {
+            const itemMenuStatus = <LinkStatus>{ text: t('New'), color: 'success' }
+            return { ...innerItem, status: itemMenuStatus }
+          }
```
Comment (ChefMomota) [inline apps/web/src/components/Menu/hooks/useMenuItems.ts:70]: remove

File: packages/uikit/src/widgets/Menu/Menu.tsx [hunk 3]
```diff
@@ -97,6 +97,21 @@ const Menu: React.FC<React.PropsWithChildren<NavProps>> = ({
...
+    () =>
+      isMobile
+        ? links.sort((a, b) => {
+            if (
+              a?.items?.[0]?.href === "https://protectors.pancakeswap.finance" ||
+              b?.items?.[0]?.href === "https://protectors.pancakeswap.finance"
+            )
+              return 1;
+            return -1;
+          })
```
Comment (ChefMomota) [inline packages/uikit/src/widgets/Menu/Menu.tsx:110]: why we need sort it?

File: apps/web/src/components/Menu/hooks/useMenuItems.ts (hunk 2 above)
Comment (Chef-Yogi) [reply apps/web/src/components/Menu/hooks/useMenuItems.ts]: done

**Pull request merged**